test/Npgsql.Tests/Types/PostgisTest.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Npgsql;
+using NpgsqlTypes;
+using NUnit.Framework;
+using NUnit.Framework.Constraints;
+
+namespace Npgsql.Tests.Types
+{
+    class PostgisTest : TestBase
+    {
+        class TestAtt
+        {
+            public IGeometry Geom;
+            public string SQL;
+        }
+
+        private Dictionary<string, TestAtt> _geoms = new Dictionary<string, TestAtt>();
+        
+        [Test]
+        [TestCase("point")]
+        [TestCase("line")]
+        [TestCase("poly")]
+        [TestCase("mpoint")]
+        [TestCase("mline")]
+        [TestCase("mpoly")]
+        [TestCase("coll")]
+        [TestCase("nestedcoll")]
+        public void PostgisTestRead(string geomIdx)
```

Comment:
Unimportant FYI: you can use [NUnit's `[TestCaseSource]` feature](http://www.nunit.org/index.php?p=testCaseSource&r=2.6.4) instead of building a dictionary etc.